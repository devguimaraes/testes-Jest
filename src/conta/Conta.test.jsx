import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Conta from "./Conta";
import { act } from "react-dom/test-utils";

describe("Componente de Conta", () => {
  test("Exibir o saldo da conta com formatacao monetário", async () => {
    act(() => {
      render(<Conta saldo={1000} />);
    });

    const saldo = screen.getByTestId("saldo-conta");

    expect(saldo.textContent).toBe("R$ 1000");
  });

  test("Verifica se a funcao realizar transacao é chamada ao clicar no botao", () => {
    const funcaoRealizarTransacao = jest.fn();

    render(<Conta saldo={1000} realizarTransacao={funcaoRealizarTransacao} />);

    const botao = screen.getByRole("button");
    const inputValor = screen.getByTestId("valor");

    expect(botao).toBeDisabled();

    fireEvent.change(inputValor, {
      target: {
        value: "1",
      },
    });
    fireEvent.click(botao);
    expect(funcaoRealizarTransacao).toHaveBeenCalled();
  });

  test("botao só pode ser clicado, quando o valor do formulario for preenchido", () => {
    render(<Conta saldo={1000} />);

    const inputValor = screen.getByTestId("valor");
    const botao = screen.getByRole("button");

    fireEvent.change(inputValor, {
      target: {
        value: "1",
      },
    });

    expect(botao).not.toBeDisabled();
  });
});

describe("Teste de Snpashot da Conta", () => {
  test("Componente Conta", () => {
    const { container } = render(<Conta />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
