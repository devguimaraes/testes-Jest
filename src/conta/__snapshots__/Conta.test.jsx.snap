// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snpashot da Conta Componente Conta 1`] = `
<div
  class="Conta-header"
>
  <h2>
    Conta
  </h2>
  <p>
    Saldo: 
    <span
      class="Saldo-valor"
      data-testid="saldo-conta"
    >
      R$ 0
    </span>
  </p>
  <form>
    <div>
      <label>
        Depósito
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="deposito"
        />
      </label>
    </div>
    <div>
      <label>
        Saque
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="saque"
        />
      </label>
    </div>
    <label>
      Valor:
    </label>
    <input
      data-testid="valor"
      name="valor"
      type="text"
      value="0"
    />
    <div>
      <button
        disabled=""
        type="submit"
      >
        Realizar operação
      </button>
    </div>
  </form>
</div>
`;
